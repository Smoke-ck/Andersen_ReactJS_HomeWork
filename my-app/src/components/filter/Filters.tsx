import { FC } from 'react';
import { connect } from 'react-redux';
import { setTodosFilter } from '../../store/actions/todosFilter';
import { IFilterAllToDo } from '../../store/reducers/todosFilter';
import './Filter.scss';
type IToDoFilter = {
    setTodosFilter: (value: string) => {}
    filter: string
}

const Filters: FC<IToDoFilter> = ({ setTodosFilter, filter }) => {
    return (
        <div className="filter">
            <button
                className={filter === 'done' ? 'filter__button filter__button--active' : 'filter__button'}
                onClick={() => setTodosFilter('done')}
            >
                Выполненные задачи
            </button>
            <button
                className={filter === 'todo' ? 'filter__button filter__button--active' : 'filter__button'}
                onClick={() => setTodosFilter('todo')}
            >
                Задачи в работе
            </button>
            <button
                className={filter === 'favorite' ? 'filter__button filter__button--active' : 'filter__button'}
                onClick={() => setTodosFilter('favorite')}
            >
                Избранные задачи
            </button>
            <button
                className={filter === 'all' ? 'filter__button filter__button--active' : 'filter__button'}
                onClick={() => setTodosFilter('all')}
            >
                Все задачи
            </button>
        </div>
    );
}

function mapStateToProps({ todosFilter }: { todosFilter: IFilterAllToDo }) {
    return { filter: todosFilter.value };
}

const mapDispatchToProps = {
    setTodosFilter,
};

export default connect(mapStateToProps, mapDispatchToProps)(Filters);