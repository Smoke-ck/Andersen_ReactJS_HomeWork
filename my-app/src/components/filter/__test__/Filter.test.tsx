import  { Filters } from "../Filters"
import  { shallow } from "enzyme";
import { shallowToJson } from 'enzyme-to-json';

const props: any = {
        setTodosFilter: jest.fn(),
        filter: "All"
    }

describe('Filters snapshot',()=>{
    let wrapper:any;
    beforeEach(() => {
         wrapper =  shallow(<Filters {...props}/>);
    });
    it("shoud render div", () => {
        const container = wrapper.find('[className="filter"]');
        expect(container).toHaveLength(1);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
      })
    it("shoud render 4 button", () => {
        const buttons = wrapper.find('[className="filter__button"]');
        expect(buttons).toHaveLength(4);
      })
      it("shoud render togleButton", () => {
        wrapper.find("button").at(0).simulate('click');
        expect(wrapper.find('[className="filter__button filter__button--active"]'));
      })
      it("shoud render togleButton", () => {
        wrapper.find("button").at(1).simulate('click');
        expect(wrapper.find('[className="filter__button filter__button--active"]'));
      })
      it("shoud render togleButton", () => {
        wrapper.find("button").at(2).simulate('click');
        expect(wrapper.find('[className="filter__button filter__button--active"]'));
      })
      it("shoud render togleButton", () => {
        wrapper.find("button").at(3).simulate('click');
        expect(wrapper.find('[className="filter__button filter__button--active"]'));
      })
});







































// describe('Selected component', () => {
//   let wrapper: any;

//   beforeEach(() => {
//     wrapper = mount(<Form{...props} />)
//   })
  
//   afterEach(() => {
//     wrapper.unmount();
//   })

//   it("component should render correctly", () => {
//     expect(wrapper).toBeTruthy();
//   })

//   // it("shoud render popup element", () => {
//   //   const popup = wrapper.find('[className="popup"]');
//   //   expect(popup).toHaveLength(1);
//   // })

//   // it("shoud render 4 button", () => {
//   //   const buttons = wrapper.find('[className="listItem__menu--button"]');
//   //   expect(buttons).toHaveLength(4);
//   // })

//   // it("shoud click for first button", () => {
//   //   wrapper.find("button").at(0).simulate('click');
//   //   expect(props.handleFavoriteToDo).toBeCalled();
//   // })

//   // it("shoud click for second button", () => {

//   //   wrapper.find("button").at(1).simulate('click');
//   //   expect(props.handleDoneToDo).toBeCalled();
//   // })

//   // it("shoud click for tird button", () => {

//   //   wrapper.find("button").at(2).simulate('click');
//   //   expect(props.handleClick).toBeCalled();
//   // })

//   // it("shoud click for fourth button", () => {

//   //   wrapper.find("button").at(3).simulate('click');
//   //   expect(props.handleToggleModal).toBeCalled();
//   // })
// })
