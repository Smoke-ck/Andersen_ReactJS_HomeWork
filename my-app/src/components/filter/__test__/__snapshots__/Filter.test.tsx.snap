// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters snapshot shoud render div 1`] = `
<div
  className="filter"
>
  <button
    className="filter__button"
    onClick={[Function]}
  >
    Выполненные задачи
  </button>
  <button
    className="filter__button"
    onClick={[Function]}
  >
    Задачи в работе
  </button>
  <button
    className="filter__button"
    onClick={[Function]}
  >
    Избранные задачи
  </button>
  <button
    className="filter__button"
    onClick={[Function]}
  >
    Все задачи
  </button>
</div>
`;
